### PR TITLE
[BUGFIX] Fix validation for `QueryAsset` with `create_temp_table: False` 

### PR DESCRIPTION
**Pull request by moatazelmohtaseb**
https://github.com/great-expectations/great_expectations/issues/8325

After investigating the error, it seems to be an issue with `table_column_types.py`'s method `_get_sqlalchemy_column_metadata` .
Tested the change on a custom QueryAsset with `create_temp_table` being either `True` or `False` 

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

**Comment by Shinnnyshinshin** [comment]
@moatazelmohtaseb thank you so much for this contribution. Excited to get this fix in, which appears to affect a number of users. 

I'm seeing the following typing error in the CI/CD. Could you have a look? 

```
Run invoke type-check --ci --pretty
  mypy cache /home/runner/work/great_expectations/great_expectations/.mypy_cache
mypy --install-types --non-interactive --pretty --warn-unused-ignores
great_expectations/expectations/metrics/table_metrics/table_column_types.py:94: error:
Unused "type: ignore" comment  [unused-ignore]
            table_selectable: str = (  # type: ignore[no-redef]
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
great_expectations/expectations/metrics/table_metrics/table_column_types.py:99: error:
Name "table_selectable" already defined on line 94  [no-redef]
            table_selectable: sqlalchemy.TextClause = batch_data.selectabl...
            ^
Found 2 errors in 1 file (checked 557 source files)
```

**Review by Shinnnyshinshin — APPROVED** [review]
Looks great. We were able to take care of some of the typing issues on our end, so we should be good to go :)

File: great_expectations/expectations/metrics/table_metrics/table_column_types.py [hunk 1]
```diff
@@ -89,16 +91,20 @@ def _spark(  # noqa: PLR0913
...
     ):
-        table_selectable: sqlalchemy.TextClause = batch_data.selectable
+        table_selectable = batch_data.selectable
         schema_name = None
     else:
-        table_selectable: str = (  # type: ignore[no-redef]
-            batch_data.source_table_name or batch_data.selectable.name
-        )
+        table_selectable = batch_data.source_table_name or batch_data.selectable.name
         schema_name = batch_data.source_schema_name or batch_data.selectable.schema
```
Comment (moatazelmohtaseb) [inline great_expectations/expectations/metrics/table_metrics/table_column_types.py:108]: I've tested the failing scenario (using a custom query asset) and it showed that  `batch_data.selectable` is an instance of  `sqlalchemy.Subquery` and threw the same error, but it was resolved when I used the following logic in the `else` block:
```
        table_selectable = batch_data.selectable
        schema_name = None
```

Comment (moatazelmohtaseb) [reply great_expectations/expectations/metrics/table_metrics/table_column_types.py:108]: @Shinnnyshinshin

**Pull request merged**